ports/esp32/machine_pwm.c

```diff
@@ -213,38 +365,93 @@ STATIC mp_obj_t mp_machine_pwm_make_new(const mp_obj_type_t *type,
 }
 
 STATIC void mp_machine_pwm_deinit(machine_pwm_obj_t *self) {
-    int chan = self->channel;
+    int chan = CHANNEL_IDX(self->mode, self->channel);
 
     // Valid channel?
-    if ((chan >= 0) && (chan < LEDC_CHANNEL_MAX)) {
+    if ((chan >= 0) && (chan < PWM_CHANNEL_MAX)) {
+        // Clean up timer if necessary
+        if (!is_timer_in_use(chan, chans[chan].timer_idx)) {
+            check_esp_err(ledc_timer_rst(self->mode, self->timer));
+            // Flag it unused
+            timers[chans[chan].timer_idx].freq_hz = -1;
+        }
+
         // Mark it unused, and tell the hardware to stop routing
-        chan_gpio[chan] = -1;
-        ledc_stop(PWMODE, chan, 0);
-        self->active = 0;
+        check_esp_err(ledc_stop(self->mode, chan, 0));
+        // Disable ledc signal for the pin
+        //gpio_matrix_out(self->pin, SIG_GPIO_OUT_IDX, false, false);
+        if (self->mode == LEDC_LOW_SPEED_MODE) {
+            gpio_matrix_out(self->pin, LEDC_LS_SIG_OUT0_IDX + self->channel, false, true);
+        } else {
+            gpio_matrix_out(self->pin, LEDC_HS_SIG_OUT0_IDX + self->channel, false, true);
```

Comment:
It looks like only the original esp32 has a high-speed mode, so the build fails on C3/S2/S3.  I'm not exactly sure how to deal with this, maybe just disallow high-speed mode on those SoCs that don't support it?

Reply:
I hope https://github.com/micropython/micropython/pull/7817/commits/6401ec34c719c2592f12c3933043374405473566
fix the high-speed mode issue.